tests/Content/ContentStorageHandlerTest.php

```diff
@@ -329,6 +329,64 @@ public function testDynamicVersionsForUser()
 		$this->assertSame(VersionId::PUBLISHED, $versions[1]->value());
 	}
 
+	/**
+	 * @covers ::from
+	 */
+	public function testFrom()
+	{
+		$this->setUpMultiLanguage();
+
+		$handlerA = new PlainTextContentStorageHandler(
+			model: $this->model
+		);
+
+		$versionPublished = VersionId::published();
+		$versionChanges   = VersionId::changes();
+
+		$en = $this->app->language('en');
+		$de = $this->app->language('de');
+
+		// create all possible versions
+		$handlerA->create($versionPublished, $en, $publishedEN = [
+			'title' => 'Published EN'
+		]);
+
+		$handlerA->create($versionPublished, $de, $publishedDE = [
+			'title' => 'Published DE'
+		]);
+
+		$handlerA->create($versionChanges, $en, $changesEN = [
+			'title' => 'Changes EN'
+		]);
+
+		$handlerA->create($versionChanges, $de, $changesDE = [
+			'title' => 'Changes DE'
+		]);
+
+		// create a new handler with all the versions from the first one
+		$handlerB = TestContentStorageHandler::from($handlerA);
+
```

Comment:
To ensure that indeed a new instance was created:

```suggestion

		$this->assertNotSame($handlerA, $handlerB);

```